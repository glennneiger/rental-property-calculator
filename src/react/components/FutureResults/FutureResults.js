import * as React from 'react';
import PropTypes from 'prop-types';

import * as css from './collapsibleEditor.css';
const css = require('./futureResults.css');
import YearResult from '../YearResult';

const FutureResults = ({
  yearsForResults
}) => {
  return (
    <div className={css.futureResults}>
      {yearsForResults.length === 1
        ? <p className={css.enterAmortization}>
          Enter an amortization period to see your results.
        </p>
        : yearsForResults.map(year => (
          <YearResult
            key={year}
            year={year}
          />
        ))}
    </div>
  );
};

// FutureResults.propTypes = {
//   yearsForResults: PropTypes.array.isRequired
// };

export default FutureResults;