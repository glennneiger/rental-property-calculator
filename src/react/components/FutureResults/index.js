import FutureResults from './FutureResults';

import { connect } from 'react-redux';

import { getYearsForResults } from '../../../utils/resultsUtils';
import {
  INPUT_ID_AMORTIZATION_PERIOD,
  TITLE_INITIAL_PURCHASE
} from '../../../constants';

export const mapStateToProps = state => ({
  yearsForResults: getYearsForResults(
    state.calculator[TITLE_INITIAL_PURCHASE].inputs[INPUT_ID_AMORTIZATION_PERIOD]
  )
});

export default connect(
  mapStateToProps
)(FutureResults);