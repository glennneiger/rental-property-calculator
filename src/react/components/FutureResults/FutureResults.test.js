import React from 'react';
import { shallow } from 'enzyme';
import { expect as chaiExpect } from 'chai';
import ShallowRenderer from 'react-test-renderer/shallow';

import FutureResults from './FutureResults';

describe('<FutureResults />', () => {
  const mockSingleYearForResult = [1];
  const mockYearsForResults = [1, 5, 10, 15, 20, 25, 26];
  const mockYearsForResultsLength = mockYearsForResults.length;

  it('renders correctly with more than 1 year for result', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <FutureResults yearsForResults={mockYearsForResults} />
    );

    const tree = renderer.getRenderOutput();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a single year for result', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <FutureResults yearsForResults={mockSingleYearForResult} />
    );

    const tree = renderer.getRenderOutput();

    expect(tree).toMatchSnapshot();
  });

  it('has correct className for styling', () => {
    const wrapper = shallow(
      <FutureResults yearsForResults={mockYearsForResults} />
    );

    chaiExpect(wrapper).to.have.className('futureResults');
  });

  it('renders a year result for each member of yearsForResults', () => {
    const wrapper = shallow(
      <FutureResults yearsForResults={mockYearsForResults} />
    );

    chaiExpect(wrapper.find('p')).to.have.length(0);
    chaiExpect(wrapper.children())
      .to.have.length(mockYearsForResultsLength);
  });

  it('renders a single child paragraph if only given 1 year for result', () => {
    const wrapper = shallow(
      <FutureResults yearsForResults={mockSingleYearForResult} />
    );

    chaiExpect(wrapper.find('p')).to.have.length(1);
    chaiExpect(wrapper.children()).to.have.length(1);
  });

  it('has proper className for styling on single child paragraph', () => {
    const wrapper = shallow(
      <FutureResults yearsForResults={mockSingleYearForResult} />
    );

    chaiExpect(wrapper.find('p').at(0)).to.have.className('enterAmortization');
  });
});