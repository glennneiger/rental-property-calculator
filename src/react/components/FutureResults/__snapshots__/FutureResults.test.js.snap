// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FutureResults /> renders correctly with a single year for result 1`] = `
<div
  className="futureResults"
>
  <p
    className="enterAmortization"
  >
    Enter an amortization period to see your results.
  </p>
</div>
`;

exports[`<FutureResults /> renders correctly with more than 1 year for result 1`] = `
<div
  className="futureResults"
>
  <Connect(YearResult)
    year={1}
  />
  <Connect(YearResult)
    year={5}
  />
  <Connect(YearResult)
    year={10}
  />
  <Connect(YearResult)
    year={15}
  />
  <Connect(YearResult)
    year={20}
  />
  <Connect(YearResult)
    year={25}
  />
  <Connect(YearResult)
    year={26}
  />
</div>
`;
